File: javascript/client/shop-js/src/models/purchase.js
export class Purchase {
    id_product;
    id_user;
    price;
    quantity;
    purchase_date;
    delivery_type;
    payment_type;

    constructor(
        id_product,
        id_user,
        price,
        quantity,
        purchase_date,
        delivery_type,
        payment_type
    ) {
        this.id_product = id_product;
        this.id_user = id_user;
        this.price = price;
        this.quantity = quantity;
        this.purchase_date = purchase_date;
        this.delivery_type = delivery_type;
        this.payment_type = payment_type;
    }
}

File: lua/src/lua/models/Purchase.lua
local Model = require("lapis.db.model").Model

local Purchase = Model.extend("purchase", {
    columns = {
        { "id", type = "serial" },
        { "id_product", type = "number" },
        { "id_user", type = "number" },
        { "price", type = "number" },
        { "quantity", type = "number" },
        { "purchase_date", type = "string" },
        { "delivery_type", type = "number" },
        { "payment_type", type = "number" }
    }
})

return Purchase